packages/@azure/servicebus/data-plane/test/stress/messageRandomDisposition.ts

```diff
@@ -0,0 +1,139 @@
+import { ServiceBusClient, SendableMessageInfo, OnMessage, OnError, delay } from "../../lib";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 5 * 12 * 24 * 7; // 1 week
+
+const messagesToProcess: Set<number> = new Set<number>();
+const messageAbandonedMap: { [key: number]: number } = {};
+
+let abandonCount = 0;
+let completeCount = 0;
+let deadletterCount = 0;
+let deferCount = 0;
+
+let msgId = 1;
+
+let snapshotIntervalID: any;
+
+let isJobDone = false;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 5000); // Every 5 seconds
+  sendMessages();
+  receiveMessages();
+}
+
+async function sendMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+  try {
+    const sender = client.createSender();
+
+    while (!isJobDone) {
+      const message: SendableMessageInfo = {
+        messageId: msgId,
+        body: "test",
+        label: `${msgId}`
+      };
+      messageAbandonedMap[msgId] = 0;
+      messagesToProcess.add(msgId);
+      msgId++;
+      await sender.send(message);
+      await delay(2000); // Throttling send to not increase queue size
+    }
+  } finally {
+    client.close();
+    ns.close();
+  }
+}
+
+async function receiveMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+
+  try {
+    const receiver = client.createReceiver();
+    const onMessageHandler: OnMessage = async (brokeredMessage) => {
+      const receivedMsgId = brokeredMessage.messageId;
+
+      if (typeof receivedMsgId !== "number") {
+        throw new Error("MessageId is corrupt or is of unexpected type");
+      }
+
+      const seed = Math.floor((Math.random() * 10) % 4);
+
+      switch (seed) {
+        case 0: {
+          const currCount = messageAbandonedMap[receivedMsgId];
+          if (currCount === 10) {
+            abandonCount++;
+            if (messagesToProcess.has(receivedMsgId)) {
+              messagesToProcess.delete(receivedMsgId);
+            }
+          }
+          messageAbandonedMap[receivedMsgId] = currCount + 1;
+          brokeredMessage.abandon();
```

Comment:
I don't think so, but it would depend on what are we looking to capture?

Reply:
It was tricky dealing with abandoning of message in context of random disposition.
The abandonCount is 0 because no message stays in queue long enough to hit delivery count of 10. And how many times a message gets abandoned and gets it's delivery count incremented is arbitrary.
The idea we discussed was that the random disposition gets evenly spread out on the types of message disposition operations over rhea link equally.

Reply:
In that case, why check if the count is 10? Should that code be removed?

Reply:
This was to keep track of the precise number of messages in flight.